tests/Core/Config/SniffsExcludeArgsTest.php

```diff
@@ -0,0 +1,199 @@
+                'message'  => sprintf($messageTemplate, 'Standard.Category'),
+            ];
+
+            // An error-code is not a valid sniff.
+            $data[$argument.'; error-code'] = [
+                'argument' => $argument,
+                'value'    => 'Standard.Category',
+                'message'  => sprintf($messageTemplate, 'Standard.Category'),
+            ];
+
+            // Only the first error is reported.
+            $data[$argument.'; two errors'] = [
+                'argument' => $argument,
+                'value'    => 'StandardOne,StandardTwo',
+                'message'  => sprintf($messageTemplate, 'StandardOne'),
+            ];
+            $data[$argument.'; valid followed by invalid'] = [
+                'argument' => $argument,
+                'value'    => 'StandardOne.Category.Sniff,StandardTwo.Category',
+                'message'  => sprintf($messageTemplate, 'StandardTwo.Category'),
+            ];
+        }//end foreach
+
+        return $data;
+
+    }//end dataInvalidSniffs()
+
+
+    /**
+     * Ensure that the valid data does not throw an exception, and the value is stored.
+     *
+     * @param string $argument 'sniffs' or 'exclude'.
+     * @param string $value    List of sniffs to include or exclude.
+     *
+     * @return       void
+     * @dataProvider dataValidSniffs
+     */
+    public function testValid($argument, $value)
+    {
+        $config = new ConfigDouble();
+        $config->processLongArgument($argument.'='.$value, 0);
+
+        $this->assertSame(explode(',', $value), $config->$argument);
+
+    }//end testValid()
+
+
+    /**
+     * Data provider for testValid().
+     *
+     * @see    self::testValid()
+     * @return array<array<string>>
+     */
+    public static function dataValidSniffs()
+    {
+        $arguments = [
+            'sniffs',
+            'exclude',
+        ];
+        $data      = [];
+
+        foreach ($arguments as $argument) {
+            $data[$argument.'; one valid sniff']  = [
+                'argument' => $argument,
+                'value'    => 'Standard.Category.Sniff',
+            ];
+            $data[$argument.'; two valid sniffs'] = [
+                'argument' => $argument,
+                'value'    => 'StandardOne.Category.Sniff,StandardTwo.Category.Sniff',
+            ];
+        }
+
+        return $data;
+
+    }//end dataValidSniffs()
+
+
+    /**
+     * Ensure that only the first argument is processed and others are ignored.
+     *
+     * @param string $argument 'sniffs' or 'exclude'.
+     *
+     * @return       void
+     * @dataProvider dataOnlySetOnce
+     */
+    public function testOnlySetOnce($argument)
+    {
+        $config = new ConfigDouble();
+        $config->processLongArgument($argument.'=StandardOne.Category.Sniff', 0);
+        $config->processLongArgument($argument.'=StandardTwo.Category.Sniff', 0);
+        $config->processLongArgument($argument.'=Standard.AnotherCategory.Sniff', 0);
```

Comment:
This is not a realistic test case as you can't have three different arguments at offset `0` on the command-line.

You may want to have a look at my previous code suggestion for this ?

Reply:
I've seen the update. Thanks. Any particular reason why you are calling the `processLongArgument()` method from within the tests (in particular this test) instead of doing something like the below ?

```php
        $cliArgs = [
            "--$argument=StandardOne.Category.Sniff",
            "--$argument=StandardTwo.Category.Sniff",
            "--$argument=Standard.AnotherCategory.Sniff",
        ];

        $config = new Config($cliArgs);
```

Reply:
> I'm trying to target a specific code block.

Using a higher level entry point, doesn't negate that you are still targetting that same code block. The `@covers` tag protects against recording incidental code coverage and using the higher level entry point is a more solid and stable way to test the real world usage of the code.

> Should the `processLongArgument()` method be non-public?

IMO absolutely, but making it so could be a breaking change for integrators.

I intend to draw up a proposal with method visibility changes at some point, but those kind of changes will be made in the 5.0 release at the earliest.

Reply:
> > Should the `processLongArgument()` method be non-public?
> IMO absolutely

I've marked this method as `@internal` and updated the tests to avoid using it directly.

Reply:
Marking the methods as `@internal` is premature and not something which belongs in this PR.
Something like that should only be done once the visibility change proposal has been posted.

Reply:
If it's not marked as internal, then how should people know to avoid using it? I used it because it was a public method which had no note / indication that it should be avoided. Shall I revert by recent changes therefore?

Reply:
Please revert the addition of `@internal`. Other than that, any discussion about visibility changes or marking things as `@internal` should be moved to their own issues and doesn't belong in this PR.